### PR TITLE
Revise documentation

### PR DESCRIPTION
**Pull request by ESKYoung**
# Summary

Revise documentation to include further detail on the `cookiecutter` package, and how to modify the structure of `govcookiecutter`. Fixed incorrect contact email address.

Added tests to check for broken links in the Sphinx documentation - ensures future releases of `govcookiecutter` have the correct links throughout the documentation.

Added information on [`cruft`](https://github.com/cruft/cruft), which could help future projects generated from `govcookiecutter` stay up-to-date with any changes we release here.

Also refactor documentation (Markdown/reStructuredText files only, predominately excluding `CODE_OF_CONDUCT.md` and `CONTRIBUTING.md`) so that it is more accessible. Accessibility checks include:

- building documentation against [`govuk-tech-docs-sphinx-theme`](https://github.com/ukgovdatascience/govuk-tech-docs-sphinx-theme) ~~[currently private, but soon to be public repository]~~
- checking against [WAVE Web Accessibility Evaluation Tool](https://wave.webaim.org/)
- checking link test is descriptive
- checking hierarchy of page headings
- removing italics, and bold text, and only using block capitals for placeholders in code examples
- checking for accessible language
   - using [`alex.js`](https://alexjs.com/)
   - replacing instances of `click` with `select` or `choose`
   - removing latin phrases, i.e. ad hoc, via
   - using [GOV.UK inclusive language](https://www.gov.uk/government/publications/inclusive-communication/inclusive-language-words-to-use-and-avoid-when-writing-about-disability)
   - [replacing negative contractions](https://www.englishclub.com/vocabulary/contractions-negative.htm)
   - aiming not to have long sentences or paragraphs (25 words per sentence; 5 lines per paragraph)
   - checking for unique titles in documentation
   - checking diagrams and images for alt text as well as surrounding contextual text, removing diagrams/images that do not add anything to a user's understanding, removing screenshots if possible, [use accessible SVGs](https://design-system.service.gov.uk/styles/images/)
   - checking for [inaccessible formats](https://www.gov.uk/guidance/how-to-publish-on-gov-uk/accessible-pdfs)

~~Once `govuk-tech-docs-sphinx-theme` goes live (July? Preview site [here](https://ukgovdatascience.github.io/govuk-tech-docs-sphinx-theme/index.html)), I will raise a new PR separately to publish the `govcookiecutter` documentation on GitHub Pages, adding a corresponding accessibility statement.~~

@alexander-newton - for interest!

# Checklists

This pull/merge request meets the following requirements:

- [x] Code runs
- [x] Developments are **secure** and [**ethical**][data-ethics-framework]
- [x] You have made proportionate checks that the code works correctly
- [x] You have tested the build of the template
- [x] Test suite passes
- [x] [Minimum usable documentation][agilemodeling] written in the `docs` folder

Comments have been added below around the incomplete checks.

[agilemodeling]: http://agilemodeling.com/essays/documentLate.htm
[data-ethics-framework]: https://www.gov.uk/government/publications/data-ethics-framework

### CODE REVIEW

**Comment by ESKYoung** [comment]
Apologies, missed a few bullet list fixes for style formatting, hence the force push.

**Comment by ESKYoung** [comment]
[Added `govuk-tech-docs-sphinx-theme` as the Sphinx theme](https://github.com/ukgovdatascience/govuk-tech-docs-sphinx-theme/). This allows us to deploy this documentation onto GitHub Pages.

**Comment by nacnudus** [comment]
# Review progress

I ran the GitHub Action on `release` locally (instructions below), which rendered the docs.  To help check that the changes are as expected, here's a screenshot with the window narrowed to resemble a mobile device.

image

Here's how they look [currently](https://ukgovdatascience.github.io/govuk-tech-docs-sphinx-theme/#gov-uk-tech-docs-sphinx-theme):

image

## Running GitHub actions locally

Using [act](https://github.com/nektos/act):

```
# From the root directory of this repository, run the action and don't stop the container
act --reuse release
# Find the name of the container
docker ps
# Copy the `docs/_build` folder of rendered docs to the local system.
docker cp <name_of_container>:$(pwd)/docs/_build docs/
# Browse the docs/build/index.html in a web browser
```

**Comment by nacnudus** [comment]
### A live checklist of what I have checked, with remarks inline.

- [ ] Revise documentation to include further detail on the `cookiecutter` package, and how to modify the structure of `govcookiecutter`.
- [ ] Fixed incorrect contact email address.
- [ ] Added tests to check for broken links in the Sphinx documentation - ensures future releases of `govcookiecutter` have the correct links throughout the documentation.
- [ ] Added information on [`cruft`](https://github.com/cruft/cruft), which could help future projects generated from `govcookiecutter` stay up-to-date with any changes we release here.

Also refactor documentation (Markdown/reStructuredText files only, predominately excluding `CODE_OF_CONDUCT.md` and `CONTRIBUTING.md`) so that it is more accessible. Accessibility checks include:

- [ ] building documentation against [`govuk-tech-docs-sphinx-theme`](https://github.com/ukgovdatascience/govuk-tech-docs-sphinx-theme) ~~[currently private, but soon to be public repository]~~
- [ ] checking against [WAVE Web Accessibility Evaluation Tool](https://wave.webaim.org/)
- [ ] checking link test is descriptive
- [ ] checking hierarchy of page headings
- [ ] removing italics, and bold text, and only using block capitals for placeholders in code examples
- [ ] checking for accessible language
   - [ ] using [`alex.js`](https://alexjs.com/)
   - [ ] replacing instances of `click` with `select` or `choose`
   - [ ] removing latin phrases, i.e. ad hoc, via
   - [ ] using [GOV.UK inclusive language](https://www.gov.uk/government/publications/inclusive-communication/inclusive-language-words-to-use-and-avoid-when-writing-about-disability)
   - [ ] [replacing negative contractions](https://www.englishclub.com/vocabulary/contractions-negative.htm)
   - [ ] aiming not to have long sentences or paragraphs (25 words per sentence; 5 lines per paragraph)
   - [ ] checking for unique titles in documentation
   - [ ] checking diagrams and images for alt text as well as surrounding contextual text, removing diagrams/images that do not add anything to a user's understanding, removing screenshots if possible, [use accessible SVGs](https://design-system.service.gov.uk/styles/images/)
   - [ ] checking for [inaccessible formats](https://www.gov.uk/guidance/how-to-publish-on-gov-uk/accessible-pdfs)

~~Once `govuk-tech-docs-sphinx-theme` goes live (July? Preview site [here](https://ukgovdatascience.github.io/govuk-tech-docs-sphinx-theme/index.html)), I will raise a new PR separately to publish the `govcookiecutter` documentation on GitHub Pages, adding a corresponding accessibility statement.~~

- [x] Code runs
- [ ] Developments are **secure** and [**ethical**][data-ethics-framework]
- [ ] You have made proportionate checks that the code works correctly
- [x] You have tested the build of the template
- [ ] Test suite passes
- [ ] [Minimum usable documentation][agilemodeling] written in the `docs` folder

**Comment by nacnudus** [comment]
Long sentences (found by tokenising into paragraphs, then sentences, then words):

1. `README.md`: the first sentence under `## Acknowledgements`.
2. `docs/contributing_guide/modify_govcookiecutter.md`: sentences beginning "Whilst", and "The constant".
3. `docs/accessibility.md`: sentence beginning "We're always"
4. `CONTRIBUTING.md`: sentence beginning "We also try"
5. `CODE_OF_CONDUCT.md`: sentences beginning "Contributors to", "In the interest" (not sure that one is fixable), "Project maintainers" (both sentences), "Examples of representing", "This code of conduct applies", "Project maintainers who do not follow"
6. `{{ cookiecutter.repo_name }}/notebooks/README.md`
7. `{{ cookiecutter.repo_name }}/docs/contributor_guide/pre_commit_hooks.md`: sentences beginning, "It's strongly", "In either case", 
8. `{{ cookiecutter.repo_name }}/data/README.md`: sentence beginning, "The folder"
9. `{{ cookiecutter.repo_name }}/docs/aqa/aqa_plan.md`: sentence beginning, "This is a living"

**Review by nacnudus — APPROVED** [review]
This represents a lot of careful work, thank you @ESKYoung!

I have skim-read the diffs to check that nothing has accidentally been pasted in and committed.  Most of the changes seem to be the odd word here and there, to meet accessibility guidelines, alexjs, etc.  There are some structural changes, such as moving chunks of text into separate files -- I haven't read those carefully, so I trust that nothing has been accidentally lost in the transfer.

I ran alexjs myself.  I couldn't get the WAVE browser addin to work with the locally rendered website, and I haven't created my own repo to render it online, so I trust that WAVE's suggestions have been addressed.

For some textual checks, such as heading hierarchy and sentence length, I wrote R scripts to help, and have mentioned things that turned up.  They're pretty trivial.

So all in all, a great PR.  Approved!

**Comment by ESKYoung** [comment]
@nacnudus thanks for the thorough review!

I've made the changes in commit [`5401bad`](https://github.com/ukgovdatascience/govcookiecutter/pull/33/commits/5401badb87028bb4318546af2e6109faf1a15455), except for the following long sentences:

- `CODE_OF_CONDUCT.md`: All remain unfixed except for the "Contributors to" paragraph. The remaining are difficult to modify given the context. As we "aim" to have short sentences/paragraphs, I'm going to leave this for now.

I've also added some guidance on writing accessible documentation that captures the above checklist. Grateful if you could have a quick skim!

File: {{ cookiecutter.repo_name }}/docs/contributor_guide/writing_accessible_documentation.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  - aim not to have long paragraphs (maximum 5 lines per paragraph)
+  - check for unique titles in documentation
+  - check diagrams and images for alternative text as well as surrounding contextual
+    text
+  - remove diagrams/images that do not add anything to a user's understanding
+  - remove screenshots if possible
+  - [use accessible SVGs][govuk-design-system-images]
+  - [check for inaccessible formats][govuk-accessible-formats]
+
+This checklist created by the GDS technical writing team with help from the GDS
```
Comment (nacnudus) [inline {{ cookiecutter.repo_name }}/docs/contributor_guide/writing_accessible_documentation.md:39]:
```suggestion
This checklist was created by the GDS technical writing team with help from the GDS
```

**Comment by nacnudus** [comment]
Thanks @ESKYoung!  Just one typo, otherwise good to go.

**Comment by ESKYoung** [comment]
Thanks for your very thorough review @nacnudus! Merging! 🚀

**Pull request merged**